File: Spring Boot/Courzelo/src/main/java/org/example/courzelo/dto/AttendanceDTO.java
package org.example.courzelo.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import org.example.courzelo.models.AttendanceStatus;

import java.time.LocalDate;
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@Data
public class AttendanceDTO {
    private String studentId;
    private String studentName;
    private String courseId;
    private String className;
    private LocalDate date;
    private AttendanceStatus status;
}

File: Angular/src/app/shared/models/AttendanceDTO.ts
export interface AttendanceDTO {
    studentId: string;
    studentName: string;
    date: string;
    status: AttendanceStatus;
}

export enum AttendanceStatus {
    PRESENT = 'PRESENT',
    ABSENT = 'ABSENT',
    LATE = 'LATE',
    EXCUSED = 'EXCUSED',
    UNEXCUSED_ABSENT = 'UNEXCUSED_ABSENT'
 }
